### PR TITLE
Frontend Deployment for Add-API

### PR DESCRIPTION
**Pull request by Amokhalad**
This PR provides a web application for fetching API calls from a API documentation urls into a specified format (Option 1), to easily generate training points to train a fine-tuned LLM model in performing API calls.

Here is a short demo of the webpage: [link](https://www.loom.com/share/b3300ddbfdc24a9d85fc436e437c3bf7?sid=fa65883d-0869-4c46-b6e4-7cd89d63fb31)

### CODE REVIEW

File: add-api/client/src/components/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+const Header: React.FC = () => {
+  return (
+    <div className="text-center mb-4">
+      <h1>🦍 Gorilla: API Zoo Data Converter</h1>
+      <div className="text-center">
+        <p className="lead">Easily give your API information to Gorilla API Store</p>
```
Comment (ramanv0) [inline add-api/client/src/components/Header.tsx:8]: Maybe reword this to "Easily **add** your API information to Gorilla API Store"?

@ShishirPatil Also, should we use "Gorilla API Store" or "Gorilla API Zoo"? I think it would be good to have a consistent name.

File: apizoo/app/package-lock.json
Comment (ramanv0) [inline apizoo/app/package-lock.json:1]: This is a question of whether we want to prioritize minimizing the probability of a dependency conflict or bloating the repo with metadata, but in my PR for the API Zoo I specifically removed the `package-lock.json` to not add this huge file (12k lines!). I think the chance of dependency conflict for this simple webpage is quite low, and the `package-lock.json` file can be generated using the `package.json` file by running `npm install`. What do you think?

**Review by ramanv0 — CHANGES_REQUESTED** [review]

File: add-api/client/src/components/GitHubAuthButton.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// GitHubAuthButton.tsx
```
[the remark this answers is not in the document]
Comment (Amokhalad) [reply add-api/client/src/components/GitHubAuthButton.tsx:1]: No all of these permissions are unnecessary, I will updates this. And yea, Gorilla can take ownership, [just create a new OAuth App ](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/creating-an-oauth-app)and send me the Client ID and Client Secret, I can update the server and frontend with these new values.

[the remark this answers is not in the document]
Comment (Amokhalad) [reply add-api/client/src/components/GitHubAuthButton.tsx]: I was only able to reduce the scope of the OAuth app down to public repos. However, within this scope, the app is still authorized to perform read and write operations on:

- Code
- Issues
- Pull requests
-Wikis
- Settings
- Webhooks and services
- Deploy keys

Unfortunately, OAuth apps do not support more granular permissions for these; these permissions are bundled together with access to public repos by GitHub’s design. To address this, we could register a GitHub App for the Gorilla in the future, which has more granular permission controls, and rewrite the server to work with the Github App instead of OAuth

File: add-api/client/src/components/Examples.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        urls: ['https://example.com/api3', 'https://example.com/api4'],
+        urlsResults: {},
+    },
+    {
+        username: 'user2',
+        apiName: 'API Two',
+        urls: ['https://example.com/api3', 'https://example.com/api4'],
+        urlsResults: {},
+    },
+    // TODO: Add more examples
```
[the remark this answers is not in the document]
Comment (Amokhalad) [reply add-api/client/src/components/Examples.tsx]: Fixed, added more examples

File: apizoo/app/package-lock.json
Comment (Amokhalad) [reply apizoo/app/package-lock.json:1]: Yea that's fine we can remove it. I did get some dependency conflicts from your repo, but they can just be fixed with `npm audit fix` I guess
